### PR TITLE
fix(protect-2846): allow restore on lns+

### PR DESCRIPTION
**Pull request by stephane-lieumont-ledger**
### 📝 Description

- Allow restore flow recover with LNS+ previous version

### ❓ Context

- [JIRA PROTECT-2846](https://ledgerhq.atlassian.net/browse/PROTECT-2846)

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - 

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Review by Justkant — CHANGES_REQUESTED** [review]
Small detail on the error handling

File: apps/ledger-live-desktop/src/renderer/components/RecoverRestore/index.tsx [hunk 1]
```diff
@@ -53,6 +54,15 @@ const RecoverRestore = () => {
...
+        if (error instanceof DeviceNotOnboarded) {
+          setState({
+            isOnboarded: false,
+            isInRecoveryMode: false,
+            seedPhraseType: SeedPhraseType.TwentyFour,
+            currentOnboardingStep: OnboardingStep.NewDevice,
+            currentSeedWordIndex: 0,
+          });
+        }
         setError(error);
```
[the remark this answers is not in the document]
Comment (stephane-lieumont-ledger) [reply apps/ledger-live-desktop/src/renderer/components/RecoverRestore/index.tsx]: https://github.com/LedgerHQ/ledger-live/pull/5674/commits/85e3a01830650e2c18a5773a57a646c0934f2518

**Review by Justkant — APPROVED** [review]
LGTM

**Review by stephane-lieumont-ledger — COMMENTED** [review]
We need to investigate, how can we manage too hold device with firmware version that not recognize Recover

File: apps/ledger-live-desktop/src/renderer/components/RecoverRestore/index.tsx [hunk 2]
```diff
@@ -53,7 +55,20 @@ const RecoverRestore = () => {
...
       error: (error: Error) => {
-        setError(error);
+        if (
+          (error instanceof TransportStatusError && error.message.includes("0x6d06")) ||
+          error instanceof DeviceNotOnboarded
+        ) {
+          setState({
+            isOnboarded: false,
+            isInRecoveryMode: false,
+            seedPhraseType: SeedPhraseType.TwentyFour,
```
Comment (alexalouit) [inline apps/ledger-live-desktop/src/renderer/components/RecoverRestore/index.tsx:65]: I'm not sure to understand, why do we set a 24 words type if the device is not onbarded?

**Review by alexalouit — CHANGES_REQUESTED** [review]

File: apps/ledger-live-desktop/src/renderer/components/RecoverRestore/index.tsx [hunk 3]
```diff
@@ -53,7 +55,20 @@ const RecoverRestore = () => {
         }
       },
       error: (error: Error) => {
-        setError(error);
+        if (
+          (error instanceof TransportStatusError && error.message.includes("0x6d06")) ||
+          error instanceof DeviceNotOnboarded
```
Comment (Justkant) [inline apps/ledger-live-desktop/src/renderer/components/RecoverRestore/index.tsx:60]: I would prefer if we did an util with the code from this so we also support stax error https://github.com/LedgerHQ/ledger-live/blob/8fe0797008950ba25d966442f2767a932ebf87c1/apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx#L437-L440

**Review by Justkant — CHANGES_REQUESTED** [review]

**Review by Justkant — APPROVED** [review]

**Review by alexalouit — APPROVED** [review]

**Pull request merged**